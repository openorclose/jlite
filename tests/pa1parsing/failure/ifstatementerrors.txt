class Main {
  Void main() {
    if (a) { // empty if with no else

    }

    if (a)
      return;
    else return; // shorthand not allowed

    if 1 == 2 { //not wrapped in parens
          return 2;
    } else {
      return 1;
    }

    if (1 == 2) // forget ) and {
              return 2;
      } else {
        return 1;
      }

    if(true) {} else {} // empty elses

    if (true) {readln(a);} else println(a); // if has {}, else no {}
    if (true) readln(a); else { println(a);} // if no {}, else has {}

    if (readln(a)) {readln(a);} else { println(a);} // if condition is not an expression3

    if (readln(a)) readln(a); // sadly most stuff breaks down once there are multiple errors ):

    // regardless, if we assume users don't commit more than one if statement mistake at a time it should be fine

    // kidding of course it's not fine. oh well.
  }
}