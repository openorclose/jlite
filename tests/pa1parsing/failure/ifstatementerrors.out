Error in line 7, column 5 : Syntax error
instead expected token classes are []
On line 3, column 9:
Error : If part of if statement must have at least one statement.

Error in line 8, column 7 : Syntax error
instead expected token classes are [error, LBRACE]
On line 7, column 9:
Error : Both the if and else part of an if statement require curly braces.

Error in line 11, column 8 : Syntax error
instead expected token classes are [error, LPAREN]
On line 11, column 5:
Error : Condition for if statement should be wrapped in parentheses.

Error in line 18, column 15 : Syntax error
instead expected token classes are [error, LBRACE]
On line 17, column 14:
Error : If part of if statement requires curly braces.

Error in line 23, column 17 : Syntax error
instead expected token classes are [error]
On line 23, column 8:
Error : If part of if statement must have at least one statement.

Error in line 26, column 5 : Syntax error
instead expected token classes are [error]
On line 25, column 9:
Error : The else part of an if statement requires curly braces.

Error in line 26, column 15 : Syntax error
instead expected token classes are [error, LBRACE]
On line 26, column 9:
Error : If part of if statement requires curly braces.

Error in line 28, column 9 : Syntax error
instead expected token classes are [error, THIS, NEW, NULL, MINUS, LPAREN, INTEGERLITERAL, STRINGLITERAL, ID, TRUE, FALSE, NOT]
On line 28, column 8:
Error : If statement should contain an expression3.

Error in line 30, column 9 : Syntax error
instead expected token classes are [error, THIS, NEW, NULL, MINUS, LPAREN, INTEGERLITERAL, STRINGLITERAL, ID, TRUE, FALSE, NOT]
Error : Couldn't repair and continue parse
